### PR TITLE
Analyzer improve subqueries representation

### PR DESCRIPTION
**Pull request by kitaisreal**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

File: src/Analyzer/QueryTreeBuilder.cpp [hunk 1]
```diff
@@ -676,14 +673,24 @@ QueryTreeNodePtr QueryTreeBuilder::buildJoinTree(const ASTPtr & tables_in_select
...
+                    }
+                    else if (table_expression_modifiers->hasSampleSizeRatio())
+                    {
+                        table_expression_modifiers_error_message += "SAMPLE";
+                    }
+
+                    throw Exception(ErrorCodes::UNSUPPORTED_METHOD,
+                        "Table expression modifiers {} are not supported for subquery {}",
+                        table_expression_modifiers_error_message,
+                        node->formatASTForErrorMessage());
```
Comment (novikd) [inline src/Analyzer/QueryTreeBuilder.cpp:693]: Same code as in `QueryAnalysisPass.cpp`. Maybe extract it to a separate function to reduce copy-paste?

File: src/Analyzer/UnionNode.cpp [hunk 2]
```diff
@@ -30,11 +30,18 @@ namespace DB
...
 
-UnionNode::UnionNode()
+UnionNode::UnionNode(SelectUnionMode union_mode_)
     : IQueryTreeNode(children_size)
+    , union_mode(union_mode_)
 {
+    if (union_mode == SelectUnionMode::UNION_DEFAULT ||
+        union_mode == SelectUnionMode::EXCEPT_DEFAULT ||
+        union_mode == SelectUnionMode::INTERSECT_DEFAULT)
+        throw Exception(ErrorCodes::BAD_ARGUMENTS, "UNION mode must be normalized");
```
Comment (novikd) [inline src/Analyzer/UnionNode.cpp:43]: Let's add to the message what union mode we received.

**Review by novikd — APPROVED** [review]

File: src/Analyzer/QueryTreeBuilder.cpp (hunk 1 above)
Comment (kitaisreal) [reply src/Analyzer/QueryTreeBuilder.cpp]: Updated.

File: src/Analyzer/UnionNode.cpp (hunk 2 above)
Comment (kitaisreal) [reply src/Analyzer/UnionNode.cpp]: Updated.

**Pull request merged**